transformations/summarization_transformation/transformation.py

```diff
@@ -0,0 +1,188 @@
+        results = []
+        verbs = [tok for tok in tokens if tok.pos_ == "VERB" or tok.pos_ == "AUX" or tok.dep_ == "aux" ]
+        roots = [tok for tok in tokens if tok.dep_ == "ROOT"]
+        checks = [0 for tok in verbs if tok.dep_ =='ROOT']
+        if 0 in checks:
+            for v in verbs:
+                subs, verbNegated = self.getAllSubs(v)
+                if len(subs) > 0:
+                    v, objs = self.getAllObjs(v)
+                    for sub in subs:
+                        for obj in objs:
+                            objNegated = self.isNegated(obj)
+                            results.append(sub.text +' '+ ('!' + v.text if verbNegated or objNegated else v.text)+' '+ obj.text)
+            # print(', '.join(results))
+
+        else:
+            for root in roots:
+                v = None
+                subs, rootNegated = [root], self.isNegated(root)
+                if len(subs) > 0:
+                    for term in root.rights:
+                        if term.pos_ in ['VERB', 'AUX']:
+                            v = term
+                    if v is not None:
+                        v, objs = self.getAllObjs(v)
+                        for sub in subs:
+                            for obj in objs:
+                                objNegated = self.isNegated(obj)
+                                results.append(sub.text +' '+ ('!' + v.text if rootNegated or objNegated else v.text)+' '+ obj.text)
+                    else:
+                        continue
+
+        final_results = [', '.join(results)]
+        return final_results
+    
+    def getSubsFromConjunctions(self, subs):
+        moreSubs = []
+        for sub in subs:
+            # rights is a generator
+            rights = list(sub.rights)
+            rightDeps = {tok.lower_ for tok in rights}
+            if "and" in rightDeps:
+                moreSubs.extend([tok for tok in rights if tok.dep_ in self.dict['SUBJECTS'] or tok.pos_ == "NOUN"])
+                if len(moreSubs) > 0:
+                    moreSubs.extend(self.getSubsFromConjunctions(moreSubs))
+        return moreSubs
+
+    def isNegated(self, tok):
+        negations = {"no", "not", "n't", "never", "none"}
+        for dep in list(tok.lefts) + list(tok.rights):
+            if dep.lower_ in negations:
+                return True
+        return False
+
+    def getObjsFromPrepositions(self, deps):
+        objs = []
+        for dep in deps:
+            if dep.pos_ == "ADP" and dep.dep_ == "prep":
+                objs.extend([tok for tok in dep.rights if tok.dep_  in self.dicts['OBJECTS'] or (tok.pos_ == "PRON" and tok.lower_ == "me")])
+                potentialNewVerb, potentialNewObjs = self.getObjFromXComp(dep.rights)
+                if potentialNewVerb is not None and potentialNewObjs is not None and len(potentialNewObjs) > 0:
+                    objs.extend(potentialNewObjs)
+        return objs
+
+    def getObjFromXComp(self, deps):
```

Comment:
Adding comments to your code, especially to point out the role of each of the functions you have will be helpful.

Reply:
Thanks for the suggestions! Added the comments for the functions and updated.